### PR TITLE
feat: zadanie 2

### PR DESCRIPTION
**Pull request by robuspospolitus**

### CODE REVIEW

**Review by milenakulik — COMMENTED** [review]
Cześć, jeśli chodzi o pierwszą część, to podoba mi się, że sprawdzasz korelację pierwszej litery kabiny z klasami. Można rozważyć utworzenie kolumny z wartościami mówiącymi o tym, czy w zbiorze początkowym osoba miała przypisany numer kabiny (Has_Cabin) - okazuje się, że jeśli nie miała, jej szanse na przeżycie maleją. Do decision trees nie potrzebujesz skalować danych, ale nie powinno to przeszkadzać. Myślę, że dobrze byłoby przetestować też model KNN na zbiorze testowym i porównać wyniki. Aby uniknąć data leakage, statystyki do imputacji braków powinny być wyliczane na X_train, a następnie zastosowane na zbiorze treningowym i testowym.

**Comment by GregW04** [comment]
Twoje review wylądowało
GENERAL COMMENT:
Praca spoko, lecz trochę uboga :) . Szkoda bo zadanko fajne do poćwiczenia i ekspyrementowania. 
Ogólnie dość duży błąd przy skalowaniu, PR zakcpetuje fo fixie z dołu (komentarz pod spodem tego), ale dodatków radziłbym scaler naprawić i sprawdzić czy wynik się polepszy/pogorszy.

NA PLUS:
- Użycie modelu baseline'owego
- Użycie GridSearchCV
- wizualiazcje parametrów do wartości metryki
- macierz pomyłek, dobra argumentacja, rozsądny wybór F1-score jako czołowej metryki
- brak wycieków, oraz poprawne użycie test_splitu (na samym końcu)

NA MINUS:
- Złe użycie scalera -> na całej macierzy X (w tym kolumnach kategorialnych zaenkodowanych) -> Powinno być tylko na kolumnach numerycznych
- Zły dobór scalera (użycie StandardScalera jest błędne w tym przypadku) -> odsyłam do wykładu 1
- Tylko 2 modele (można było jeszcze zbadać LogisticRegression/ inne warianty modelów liniowych, SVM, Lasy Losowe, Drzewa wzmacniane gradientowo)


WOLNE MYŚLI:
-GridSearch jest dość słabym sposobem hyperparamater tuningu (już lepszy jest RandomSearch). Ja personalnie polecam Optuna (warto poczytać jak działa).
- Brak głębszego badania potencjalnego overfittingu/underfittingu (np. poprzez podział na train-val-test split i krzywe uczenia)
- Trafnie stwierdzone, że drzewa podatne na overiftting brak sprawdzenia modeli rozwiązujących ten problem (np. Lasy Losowe, Drzewa wzmacniane gradientowo)
-  Brak użycia metryki ROC-AUC oraz Accuracy (do porównania)

**Review by GregW04 — APPROVED** [review]

**Review by GregW04 — CHANGES_REQUESTED** [review]
shoutout dla @RakPiotr za zauważenie tego błędu w innym review. Tutaj też to jest do poprawy https://github.com/Solvro/ml-wakacyjne-wyzwanie-2026/pull/95#pullrequestreview-4984119687

**Comment by sz3kz** [comment]
Moje 3 grosze:
* Moja analiza `Fare` wykazała jeden wielki outlier (+$500). Jeśli nie transformowałeś tej kolumny na przykład logarytmicznie, to nie powinieneś używać `StandardScaler` do skalowania danych, gdyż jest on czuły na outliery. W tej sytuacji proponowałbym `RobustScaler`.
* `Cabin` zauważyłeś że ma duże braki(77%). Jeśli nie masz ochoty uzupełniać ponad połowy zbioru wartościami *null*owymi typu "O" jak Other, to warto przemyśleć zrobienie prostej transformacji `Cabin` -> `Has_Cabin`, binarnej wartości czy pasażer miał jakąś znaną nam kabinę.
* Wydaje mi się, że `PassengerId` nie było usunięte ze zbioru
* Uważam że po utworzeniu `FamilySize` z wartości `SibSp` i `Parch`, kolumny pierwotne powinny być usunięte
* Osobiście użyłbym **Optuny** do znajdywania wartości hiperparametrów.
* Widzę że seeding stanów / wartości losowych jest, very good
* Baseline w postaci `DummyClassifier` to +15 social credit
* Bardzo podobała mi się graficzna analiza optymalizacji hiperparametrów + wartości metryk modelu
